bug: bindings/c  Bump CMake version to allow successful builds of the binary

### Describe the bug

under `bindings/c` the `CMakeLists.txt` file mentions the following CMake policy `cmake_policy(SET CMP0135 NEW)` but the minimum version of CMake is set to `3.22` whereas the policy used was introduced in version `3.24`

### Steps to Reproduce

Go to `bindings/c` and follow the instructions in `CONTRIBUTING.md`. Running cmake will produce this error 

```shell
 cmake ..
CMake Error at CMakeLists.txt:19 (cmake_policy):
  Policy "CMP0135" is not known to this version of CMake.
```

### Expected Behavior

The build should succeed without any errors given the minimum version of CMake installed is correct

### Additional Context

https://cmake.org/cmake/help/latest/policy/CMP0135.html

### Are you willing to submit a PR to fix this bug?

- [X] Yes, I would like to submit a PR.

--- a/bindings/c/CMakeLists.txt
+++ b/bindings/c/CMakeLists.txt
@@ -15,7 +15,7 @@
 # specific language governing permissions and limitations
 # under the License.
 
-cmake_minimum_required(VERSION 3.22)
+cmake_minimum_required(VERSION 3.24)
 cmake_policy(SET CMP0135 NEW)
 
 project(opendal-c)

